server/channels/store/sqlstore/reaction_store.go

```diff
@@ -211,57 +211,50 @@ func (s SqlReactionStore) PermanentDeleteByUser(userId string) error {
 	now := model.GetMillis()
 	postIds := []string{}
 
-	err := s.GetReplicaX().Select(&postIds, "SELECT PostId FROM Reactions WHERE UserId = ?", userId)
+	txn, err := s.GetMasterX().Beginx()
 	if err != nil {
-		return errors.Wrapf(err, "failed to get Reactions with userId=%s", userId)
+		return err
 	}
 
-	batchIds := [][]string{}
-	batchSize := 1000
-	resultsSize := len(postIds)
-	for i := 0; i < resultsSize; i += batchSize {
-		end := i + batchSize
+	err = txn.Select(&postIds, "SELECT PostId FROM Reactions WHERE UserId = ?", userId)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get Reactions with userId=%s", userId)
+	}
 
-		if end > resultsSize {
-			end = resultsSize
-		}
+	query := s.getQueryBuilder().
+		Delete("Reactions").
+		Where(sq.And{
+			sq.Eq{"PostId": postIds},
+			sq.Eq{"UserId": userId},
+		})
 
-		batchIds = append(batchIds, postIds[i:end])
+	_, err = txn.ExecBuilder(query)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete reactions with userId=%s", userId)
 	}
+	if err = txn.Commit(); err != nil {
+		return err
+	}
+	finalizeTransactionX(txn, &err)
```

Comment:
Is it a problem if we return from any of the earlier if conditions, and the `finalizeTransactionX` won't run? We could just refactor them to an internal function to be able to use the `defer` clause. Up to you.

Reply:
Refactored to an internal function